templates/demo/app/components/storage.js

```diff
@@ -0,0 +1,174 @@
+import EmbarkJS from 'Embark/EmbarkJS';
+import React from 'react';
+import { Alert, Form, FormGroup, FormControl, HelpBlock, Button } from 'react-bootstrap';
+ 
+class Storage extends React.Component {
+
+    constructor(props) {
+      super(props);
+      
+      this.state = {
+          textToSave: 'hello world!',
+          generatedHash: '',
+          loadText: '',
+          storedText: '',
+          fileToUpload: null,
+          fileHash: '',
+          imageToDownload: '',
+          url: '',
+          logs: []
+      };
+
+      this.setText = this.setText.bind(this);
+      this.loadHash = this.loadHash.bind(this);
+      this.uploadFile = this.uploadFile.bind(this);
+      this.handleFileUpload = this.handleFileUpload.bind(this);
+      this.loadFile = this.loadFile.bind(this);
+    }
+
+    handleChange(e, name){
+        this.state[name] = e.target.value;
+        this.setState(this.state);
+    }
+
+    handleFileUpload(e){
+        this.setState({ fileToUpload: [e.target] });
+    }
+
+    _addToLog(txt){
+        this.state.logs.push(txt);
+        this.setState({logs: this.state.logs});
+    }
+
+    setText(e){
+        e.preventDefault();
+        
+        let _this = this;
+
+        EmbarkJS.Storage.saveText(this.state.textToSave)
+            .then(function(hash) {
+                _this.setState({
+                    generatedHash: hash,
+                    loadText: hash
+                });
+                _this._addToLog("EmbarkJS.Storage.saveText('" + _this.state.textToSave + "').then(function(hash) { })");
+                })
+            .catch(function(err) {
+                if(err){
+                    console.log("Storage saveText Error => " + err.message);
+                }
+            });
+    }
+
+    loadHash(e){
+        e.preventDefault();
+
+        let _this = this;
+
+        EmbarkJS.Storage.get(this.state.loadText)
+            .then(function(content) {
+                _this.setState({storedText: content});
+                _this._addToLog("EmbarkJS.Storage.get('" + _this.state.loadText + "').then(function(content) { })");
+            })
+            .catch(function(err) {
+                if(err){
+                    console.log("Storage get Error => " + err.message);
+                }
+            });
+    }
+
+    uploadFile(e){
+        e.preventDefault();
+
+        let _this = this;
+
+        EmbarkJS.Storage.uploadFile(this.state.fileToUpload)
+            .then(function(hash) {
+                _this.setState({
+                    fileHash: hash,
+                    imageToDownload: hash
+                });
+                _this._addToLog("EmbarkJS.Storage.uploadFile(this.state.fileToUpload).then(function(hash) { })");
+                })
+            .catch(function(err) {
+                if(err){
+                    console.log("Storage uploadFile Error => " + err.message);
```

Comment:
Maybe we should show the error with a state change for it to be clearer to the user.

Reply:
Fixed: The screen shows an error notice when there's an error when using the storage.